packages/babel-plugin-proposal-decorators/src/transformer-2022-03.ts

```diff
@@ -1020,6 +1037,186 @@ function transformClass(
   return path;
 }
 
+function isProtoSetter(
+  node: t.Identifier | t.StringLiteral | t.BigIntLiteral | t.NumericLiteral,
+) {
+  return node.type === "Identifier"
+    ? node.name === "__proto__"
+    : node.value === "__proto__";
+}
+
+function isDecorated(node: t.Class | ClassDecoratableElement) {
+  return node.decorators && node.decorators.length > 0;
+}
+
+function shouldTransformElement(node: ClassElement) {
+  switch (node.type) {
+    case "ClassAccessorProperty":
+      return true;
+    case "ClassMethod":
+    case "ClassProperty":
+    case "ClassPrivateMethod":
+    case "ClassPrivateProperty":
+      return isDecorated(node);
+    default:
+      return false;
+  }
+}
+
+function shouldTransformClass(node: t.Class) {
+  return (
+    isDecorated(node) ||
+    node.body.body.some(node => shouldTransformElement(node))
+  );
+}
+
+// Todo: unify name references logic with helper-function-name
+function NamedEvaluationVisitoryFactory(
```

Comment:
At first I tried to extract logic from `helper-function-name` but it turns out that helper is out-of-dated. Since any exports in helpers must await a new minor release, I end up implementing a new namedEvaluationVisitor.

Reply:
If we were to put this in helper-function-name waiting for a minor, would we need to keep both implementations or can we just update the old one?